src/utils.rs

```diff
@@ -141,8 +142,54 @@ impl Console {
     }
 
     /// Read a chunk of raw data from the console and output it.
-    pub fn read_to(&self, output: &mut dyn Write) -> NitroCliResult<()> {
+    pub fn read_to(
+        &self,
+        output: &mut dyn Write,
+        disconnect_timeout: Option<u64>,
+    ) -> NitroCliResult<()> {
+        let epoll_fd = epoll::epoll_create().map_err(|e| {
+            new_nitro_cli_failure!(
+                &format!("Failed to create epoll: {:?}", e),
+                NitroCliErrorEnum::EpollError
+            )
+        })?;
+
+        let mut epoll_event = EpollEvent::new(EpollFlags::EPOLLIN, self.fd as u64);
+        epoll::epoll_ctl(epoll_fd, EpollOp::EpollCtlAdd, self.fd, &mut epoll_event).map_err(
+            |e| {
+                new_nitro_cli_failure!(
+                    &format!("Failed to add fd to epoll: {:?}", e),
+                    NitroCliErrorEnum::EpollError
+                )
+            },
+        )?;
+
+        let mut events = [EpollEvent::empty(); 1];
+        // Disconnect timeout in microseconds
+        let mut epoll_timeout_us = match disconnect_timeout {
+            Some(time) => time as i128 * 1000 * 1000,
+            None => -1 * 1000,
```

Comment:
so, you want -1 here.
Specifying a timeout of  -1  causes
       epoll_wait() to block indefinitely, while specifying a timeout equal to
       zero cause epoll_wait() to return immediately, even if  no  events  are
       available.

Reply:
this variable is stored in microseconds, but later in the code #L180, this is conveted to milliseconds (* 1000), so after conversion, the -1000 will become -1